### ISSUE TITLE
[L10N] The new onboarding is wronlgy displayed and scrollable in RTL languages

### ISSUE THREAD

**Issue by lobontiumira** [mozilla-mobile/focus-android#6639]
## Steps to reproduce
1. Have the latest Focus Nightly installed, and with clear data.
2. Set the device to Persian.
3. Open Focus Nightly and observe the new onboarding.

### Expected behavior
The new onboarding is visible to the user on one page, nothing to scroll.

### Actual behavior
The new onboarding is scrollable, only the icons are visible, not the text.

### Device information
* Android device: Samsung Galaxy Note 8 (Android 9)
* Focus version: Focus Nightly 100.0a1 from 3/10

https://user-images.githubusercontent.com/40694158/157857437-2e4b5356-eb6a-491c-a316-5d41283bfbed.mp4

**Comment by Mugurell** [comment]
Would the same issue happen for all RTL locales? Or it's happening only for the Persian locale?

**Comment by lobontiumira** [comment]
@Mugurell yes.
I've tested also Hebrew, and Arabic.

**Comment by indurs** [comment]
@Mugurell , can I work on this issue? I can reproduce the issue @lobontiumira has reported. 
Seems like the onboarding page is also not choosing the appropriate language for the locale . For the layout I think we need to use the ConstraintLayout as the parent composable and set the appropriate constraints for its children to position according to the text direction set in the device -RTL/LTR

**Comment by indurs** [comment]
RTL

![Screenshot_۲۰۲۲۰۳۱۴-۰۹۵۵۴۷](https://user-images.githubusercontent.com/12825812/158105676-06cf6b51-0bdc-47e4-baf3-17663ea0b6d6.png)

LTR

![Screenshot_20220314-095303](https://user-images.githubusercontent.com/12825812/158105708-abf8580d-0ce7-4a34-aa66-519e6b948e70.png)

* Seems like values-ar/strings.xml does not have values for Onboarding screen . Not sure about other RTL languages.

@Composable
    fun OnboardingContent(onboardingInteractor: OnboardingInteractor) {
        Box(
            modifier = Modifier
                .background(color = focusColors.secondaryBackground)
        ){
            ConstraintLayout(modifier = Modifier.fillMaxSize().
                                        background(color = focusColors.background)

            ){
                val(onboardingLogo,onboardingTitle,onboardingDescription,
                    mozacicPrivateBrowsing, icHistory, mozacIcSettings, onboardingButton,spacer) = createRefs()

                Image(
                    painter = painterResource(R.drawable.onboarding_logo),
                    contentDescription = getString(R.string.app_name),
                    modifier = Modifier
                        .size(60.dp, 60.dp)
                        .constrainAs(onboardingLogo) {
                            start.linkTo(parent.start)
                            end.linkTo(parent.end)
                            top.linkTo(parent.top)
                            bottom.linkTo(onboardingTitle.top)
                            centerAround(
                                parent.top
                            )
                        }
                )
                Text(
                    text = stringResource(id = R.string.onboarding_title, stringResource(id = R.string.app_name)),
                    style = focusTypography.onboardingTitle,
                    modifier = Modifier
                        .padding(top = 16.dp, bottom = 20.dp)
                        .constrainAs(onboardingTitle) {
                            top.linkTo(onboardingLogo.bottom)
                            centerHorizontallyTo(parent)
                        }
                )

                Text(
                    text = stringResource(id = R.string.onboarding_description),
                    style = focusTypography.onboardingDescription,
                    modifier = Modifier
                        .padding(start = 8.dp,bottom=20.dp)
                        .constrainAs(onboardingDescription) {
                            top.linkTo(onboardingTitle.bottom)
                            centerHorizontallyTo(parent)
                        }
                )


                    KeyFeatureCard(
                        iconId = R.drawable.mozac_ic_private_browsing,
                        titleId = R.string.onboarding_incognito_title,
                        descriptionId = R.string.onboarding_incognito_description,
                        modifier = Modifier.constrainAs(mozacicPrivateBrowsing){
                            centerHorizontallyTo(parent)
                            start.linkTo(parent.start)
                            top.linkTo(onboardingDescription.bottom)
                            end.linkTo(parent.end)
                        }
                    )
                    KeyFeatureCard(
                        iconId = R.drawable.ic_history,
                        titleId = R.string.onboarding_history_title,
                        descriptionId = R.string.onboarding_history_description2,
                        modifier = Modifier.constrainAs(icHistory){
                            start.linkTo(parent.start)
                            top.linkTo(mozacicPrivateBrowsing.bottom)
                            end.linkTo(parent.end)
                            centerHorizontallyTo(parent)
                        }
                    )
                    KeyFeatureCard(
                        iconId = R.drawable.mozac_ic_settings,
                        titleId = R.string.onboarding_protection_title,
                        descriptionId = R.string.onboarding_protection_description,
                        modifier = Modifier.constrainAs(mozacIcSettings){
                            start.linkTo(parent.start)
                            top.linkTo(icHistory.bottom)
                            end.linkTo(parent.end)
                            centerHorizontallyTo(parent)
                        }
                    )
                Spacer(modifier = Modifier.height(20.dp).constrainAs(spacer){
                    start.linkTo(parent.start)
                    top.linkTo(mozacIcSettings.bottom)
                    end.linkTo(parent.end)
                    centerHorizontallyTo(parent)
                })
                Button(
                    onClick = {
                        Onboarding.finishButtonTapped.record(Onboarding.FinishButtonTappedExtra())
                        onboardingInteractor.finishOnboarding(
                            requireContext(),
                            requireComponents.store.state.selectedTabId
                        )
                    },
                    modifier = Modifier
                        .width(232.dp)
                        .height(36.dp)
                        .constrainAs(onboardingButton) {
                            top.linkTo(spacer.bottom)
                            start.linkTo(parent.start)
                            end.linkTo(parent.end)
                            bottom.linkTo(parent.bottom)
                            centerHorizontallyTo(parent)
                        }
                    ,
                    colors = ButtonDefaults.buttonColors(backgroundColor = focusColors.onboardingButtonBackground)
                ) {
                    Text(
                        text = stringResource(id = R.string.onboarding_start_browsing),
                        modifier = Modifier
                            .align(Alignment.CenterVertically),
                        style = focusTypography.onboardingButton,
                    )
                }

                createVerticalChain(onboardingLogo,onboardingTitle,onboardingDescription,
                    mozacicPrivateBrowsing, icHistory, mozacIcSettings, onboardingButton, chainStyle = ChainStyle.Packed(50f))
            }

        }

    }

**Comment by Mugurell** [comment]
@indurs 
I think you're on the right track by tweaking the constraints set in `KeyFeatureCard`. Seems like the current contraints there are causing this bug.
Not sure that we need another `ContraintLayout` as the first child of the parent `Box` though.
The proposed fix still needs some tweaking to ensure the current layout is still kept though (`KeyFeatureCard`s have to be wider):

| Current layout | Proposed layout |
| --- | --- |
| <img src="https://user-images.githubusercontent.com/11428869/158118176-3bddbb56-6eb8-42bf-b785-44b1aecb17e5.png" width="411" /> | <img src="https://user-images.githubusercontent.com/11428869/158117675-7f7064f2-6797-4130-aabc-b12d0bf3606f.png" width="500" /> |

---
> Seems like values-ar/strings.xml does not have values for Onboarding screen . Not sure about other RTL languages.

This is expected. Not an issue.

**Comment by indurs** [comment]
Thanks for the info.. Will keep working towards the proper fix

**Comment by indurs** [comment]
@Mugurell , for KeyFeatureCard composable I've used Row instead of ContraintLayout.
I didn't change the constraints and layout for OnboardingContent compossable except for the onboarding description alignment constraint.
I aligned the onboarding description to center horizontally. If it's not necessary I'll remove that constraint.

I also noted that the status bar is only visible in light theme . When dark theme is set, the status bar is not visible. Just wanted to know whether this is  the excepted behavior


```
@Composable
    fun KeyFeatureCard(iconId: Int, titleId: Int, descriptionId: Int) {
        Row(modifier = Modifier.fillMaxWidth().padding(start=16.dp)){
            Column() {
                Image(
                    painter = painterResource(iconId),
                    colorFilter = ColorFilter.tint(color = focusColors.onboardingKeyFeatureImageTint),
                    contentDescription = getString(R.string.app_name),
                )
            }
            Column(modifier = Modifier.padding(start = 16.dp)) {
                Text(
                    text = stringResource(id = titleId),
                    style = focusTypography.onboardingSubtitle
                )
                Text(
                    text = stringResource(id = descriptionId, getString(R.string.onboarding_short_app_name)),
                    style = focusTypography.onboardingDescription
                )
            }
        }
    }
```
Onboaring description horizontally and not horizontally centered

![Screenshot_20220315-064324](https://user-images.githubusercontent.com/12825812/158306814-a4f84c1a-3b1c-4949-9ed7-2d1b34231f14.png)   ![Screenshot_20220315-064250](https://user-images.githubusercontent.com/12825812/158306971-4d298687-2c2f-4aec-b16a-adda3a025e59.png)


Onboarding screen for RTL

![Screenshot_۲۰۲۲۰۳۱۵-۱۰۰۶۲۳](https://user-images.githubusercontent.com/12825812/158307840-e3e60f9a-8bd0-4c2b-9762-1e6946637c63.png)

**Comment by Mugurell** [comment]
> @Mugurell , for KeyFeatureCard composable I've used Row instead of ContraintLayout. I didn't change the constraints and layout for OnboardingContent compossable except for the onboarding description alignment constraint. 

The results look great!
Could you open a PR for the changes?

> I aligned the onboarding description to center horizontally. If it's not necessary I'll remove that constraint.

Based on [the designs from Figma](https://www.figma.com/file/ahqwL3sByHprH2Rru9R58b/Onboarding-2022?node-id=2647%3A11565) (link just for reference) this was intended to be centered. Thank you for catching that!
 
> I also noted that the status bar is only visible in light theme . When dark theme is set, the status bar is not visible. Just wanted to know whether this is the excepted behavior

Sounds like a bug. Could you open a new ticket for this?

**Comment by indurs** [comment]
Thank You @Mugurell ! I'll open a PR and will create a ticket for the bug

**Comment by lobontiumira** [comment]
Verified as fixed on the latest Focus Nightly 100.0a1 build from 3/18 with Samsung Galaxy Note 8 (Android 9), and Lenovo tablet M10 (Android 10).
I filed #6695 for the wrongly displayed dots.

![Untitled1](https://user-images.githubusercontent.com/40694158/158962218-6521b416-cd04-4916-ad92-6455bd1e688b.png)
